specificatie/BRK-Bevragen/openapi.yaml

```diff
@@ -1671,21 +1674,34 @@ components:
         identificatie:
           type: "string"
     StukHal:
-      oneOf: 
-#       - $ref: "#/components/schemas/AbstractStuk"   Nodig voor het correct genereren van code. 
-        - $ref: "#/components/schemas/AangebodenStukHal"
-        - $ref: "#/components/schemas/KadasterStukHal"
-      discriminator:
-        propertyName: stukType
+      type: object
+      description: "Een stuk is een document waaruit een wijziging in de BRK blijkt.\n
+      Er zijn 2 soorten stukken: \n
+* Een ter inschrijving in de openbare registers aangeboden stuk en \n
+* een door het Kadaster opgemaakt stuk.
+\n \n
+In een stuk is daarom óf een kadasterstuk óf een terInschrijvingAangebodenStuk opgenomen."
+      properties:
+        kadasterstuk:
+          $ref: "#/components/schemas/KadasterStuk"
+        terInschrijvingAangebodenStuk:
+          $ref: "#/components/schemas/AangebodenStuk"
```

Comment:
Dit levert nu toch een Object op met zowel een veld kadasterstuk als een veld aangebodenstuk welke in theorie beiden gevuld kunnen worden? Met de oneof constructie voorkom je dit toch juist?

Reply:
> Met de oneof constructie voorkom je dit toch juist?

Klopt, maar dat heeft nadelen op gegenereerde code. Daarom nu het voorstel om die constructie weg te laten. De provider (het Kadaster) moet zorgen dat er maar één van de twee wordt geleverd. Dat hoeft het schema m.i. niet af te dwingen. Het schema is immers gericht op de gebruiker die moet weten wat zij of hij aan gegevens kan krijgen.